### DISCUSSION TITLE
ESC and Servo

### DISCUSSION THREAD

**Discussion by zye1996** [prl-mushr/mushr#24] [Hardware]
Hi there, it is a great project and I do appreciate what you have done! I am wondering if there are specific reasons that you are not using the ESC and servo coming with the car?

**Comment by palanc** [comment]
Hi @zye1996 - thanks we're glad you like it!

The reason we don't use the stock servo is because it is not sufficiently powerful. The amount of weight added by mounting things on top of the chassis makes the stock servo move very slowly or even prevent it from moving. Also, the replacement servo uses a metal servo arm and gearing, improving its likelihood of surviving in the event of a collision.

We use the VESC instead of the stock ESC because it allows us to control the car sufficiently accurately. Essentially, if we command the car to go forwards at a speed of 1 meter per second, we can be confident that it will actually move at approximately 1 meter per second with some sufficiently small amount of error. This accuracy directly determines the quality of our odometry computations, which is important for higher level applications such as localization of the car. Our (limited) past experiences with ESCs similar to the stock ESC is that the (lack of) accuracy to which they can control the car severely degrades the quality of the odometry estimates. However we have not verified this with this particular stock ESC, and so it is possible that it could provide sufficiently accurate control - we just haven't invested the time/resources to do that evaluation.
